src/main/java/org/osc/manager/ism/api/IsmSecurityGroupInterfaceApi.java

```diff
@@ -122,12 +122,13 @@ public void updateSecurityGroupInterface(SecurityGroupInterfaceElement sgiElemen
 
     @Override
     public void deleteSecurityGroupInterface(String id) throws Exception {
-        SecurityGroupInterfaceEntity existingSgi = (SecurityGroupInterfaceEntity) getSecurityGroupInterfaceById(id);
-        if (existingSgi == null) {
-            LOG.warn(String.format("Security group interface with id %s was not found.", id));
-            return;
-        }
         this.txControl.required(() -> {
+            SecurityGroupInterfaceEntity existingSgi = (SecurityGroupInterfaceEntity) getSecurityGroupInterfaceById(id);
```

Comment:
same here. getSecurityGroupInterfaceById could throw an exception, dont put that in a transaction. Basically Validate all input before entering a transaction.